synapse/handlers/sync.py

```diff
@@ -1617,6 +1618,7 @@ async def _generate_sync_entry_for_account_data(
                 account_data_by_room,
             ) = await self.store.get_account_data_for_user(sync_config.user.to_string())
 
+            global_account_data = dict(global_account_data)
             global_account_data["m.push_rules"] = await self.push_rules_for_user(
                 sync_config.user
             )
```

Comment:
This looks like a bug. `global_account_data` comes from a cache.

Reply:
As above, I think we should just fix this separately? (I also noticed this as part of #14973.)

Reply:
PRed as https://github.com/matrix-org/synapse/pull/15047